compiler/noirc_frontend/src/hir/comptime/errors.rs

```diff
@@ -0,0 +1,54 @@
+use crate::{node_interner::DefinitionId, Type};
+use acvm::FieldElement;
+use noirc_errors::Location;
+
+use super::value::Value;
+
+/// The possible errors that can halt the interpreter.
+#[allow(unused)]
+#[derive(Debug)]
+pub(crate) enum InterpreterError {
+    ArgumentCountMismatch { expected: usize, actual: usize, call_location: Location },
+    TypeMismatch { expected: Type, value: Value, location: Location },
+    NoValueForId { id: DefinitionId, location: Location },
+    IntegerOutOfRangeForType { value: FieldElement, typ: Type, location: Location },
+    ErrorNodeEncountered { location: Location },
+    NonFunctionCalled { value: Value, location: Location },
+    NonBoolUsedInIf { value: Value, location: Location },
+    NonBoolUsedInConstrain { value: Value, location: Location },
+    FailingConstraint { message: Option<Value>, location: Location },
+    NoMethodFound { object: Value, typ: Type, location: Location },
+    NonIntegerUsedInLoop { value: Value, location: Location },
+    NonPointerDereferenced { value: Value, location: Location },
+    NonTupleOrStructInMemberAccess { value: Value, location: Location },
+    NonArrayIndexed { value: Value, location: Location },
+    NonIntegerUsedAsIndex { value: Value, location: Location },
+    NonIntegerIntegerLiteral { typ: Type, location: Location },
+    NonIntegerArrayLength { typ: Type, location: Location },
+    NonNumericCasted { value: Value, location: Location },
+    IndexOutOfBounds { index: usize, length: usize, location: Location },
+    ExpectedStructToHaveField { value: Value, field_name: String, location: Location },
+    TypeUnsupported { typ: Type, location: Location },
+    InvalidValueForUnary { value: Value, operator: &'static str, location: Location },
+    InvalidValuesForBinary { lhs: Value, rhs: Value, operator: &'static str, location: Location },
+    CastToNonNumericType { typ: Type, location: Location },
+    QuoteInRuntimeCode { location: Location },
+    NonStructInConstructor { typ: Type, location: Location },
+    CannotInlineMacro { value: Value, location: Location },
+    UnquoteFoundDuringEvaluation { location: Location },
+
+    Unimplemented { item: &'static str, location: Location },
+
+    // Perhaps this should be unreachable! due to type checking also preventing this error?
+    // Currently it and the Continue variant are the only interpreter errors without a Location field
+    BreakNotInLoop,
+    ContinueNotInLoop,
+
+    // These cases are not errors but prevent us from running more code
+    // until the loop can be resumed properly.
+    Break,
+    Continue,
```

Comment:
why don't `BreakNotInLoop`, `Break`, etc. have locations?

Reply:
`Break` and `Continue` aren't really errors, they're basically exceptions used for control flow (shh) that were originally added in the first interpreter PR - I've just done some reorganization here. The interpreter uses these to break out of loops easily without having to check for `Value::Break` or similar after every return manually. See this discussion: https://github.com/noir-lang/noir/pull/4821#discussion_r1569078422

Reply:
For `BreakNotInLoop` and `ContinueNotInLoop`, they originally didn't have any locations to give since statements didn't have locations stored with them. I changed that in this PR though in the name resolver so I'll add locations to these now